adding eeid_adder (#4)
magnoxemo/cardinal: doc/content/source/meshgenerators/ParsedElementIDMeshGenerator.md

# ParsedElementIDMeshGenerator

## Description

`ParsedElementIDMeshGenerator` is a mesh generator that wraps an existing input mesh and adds one or
more extra element integer id (eeid) to the mesh, where each eeids corresponds to user-defined values.
Clustering for mesh tally amalgamation, can benefit from it by using these eeid to label the cluster 
based on different heuristics.

!alert! warning

> >If an `extra_element_integer_names` entry already exists in the mesh, the generator **will not override** the values.

Reviewer: ```suggestion
If an `extra_element_integer_names` entry already exists in the mesh, the generator +will not+ override the values.
```

Unfortunately MOOSE uses a different markdown convention for bold text... 
See the [core extension](https://mooseframework.inl.gov/python/MooseDocs/extensions/core.html) for more of MOOSE's markdown quirks.

> >It will show a warning message and skip that field.

Reviewer: ```suggestion
It will show a warning message and skip that field.
```

> >This ensures existing mesh annotations are preserved.

Reviewer: ```suggestion
This ensures existing mesh annotations are preserved.
```

!alert-end!

## Example Input File

> This example reads a mesh file (in this case mesh_in.e) by using [FileMeshGenerator](FileMeshGenerator.md)

Reviewer: ```suggestion
This example reads a mesh file (in this case mesh_in.e) by using a [FileMeshGenerator](FileMeshGenerator.md),
```

> then the `ParsedElementIDMeshGenerator` adds three eeid **(a, b, c)** to the mesh. The values aginst eeid **(a, b, c)** will be **(-1, 44, 20)** through out the whole mesh.

Reviewer: ```suggestion
then the `ParsedElementIDMeshGenerator` adds three eeid +(a, b, c)+ to the mesh. The values of eeid +(a, b, c)+ will be +(-1, 44, 20)+ throughout the whole mesh.
```

# ParsedElementIDMeshGenerator

## Description

`ParsedElementIDMeshGenerator` is a mesh generator that wraps an existing input mesh and adds one or
more extra element integer id (eeid) to the mesh, where each eeids corresponds to user-defined values.

> Clustering for mesh tally amalgamation, can benefit from it by using these eeid to lebel the cluster 

Reviewer 2: ```suggestion
Clustering for mesh tally amalgamation, can benefit from it by using these eeid to label the cluster 
```

based on different heuristics.

> ## Input Parameters:

Reviewer 2: This block can be deleted, it will be auto-rendered by the commands I listed at the end of this file

- `extra_element_integer_names` list of the eeid names.
- `values` values for the eeid_names.
- `input` the mesh where we want to add the eeids.

> ## Warning

Reviewer 2: Instead of having a section titled `Warning`, put the warning into an alert syntax block.

https://mooseframework.inl.gov/python/MooseDocs/extensions/alert.html

If an `extra_element_integer_names` entry already exists in the mesh, the generator **does not override** the values.
It will show a warning message and skip that field. This ensures existing mesh annotations are preserved.

## Example Input File

> This example reads a mesh file (in this case mesh_in.e) by using [FileMeshGenerator](https://mooseframework.inl.gov/source/meshgenerators/FileMeshGenerator.html)

Reviewer 2: To link against other files, don't use URLs because file paths could change. Instead, use the actual filename itself. Cardinal is set up to locate the correct markdown files in its submodules.

```
[FileMeshGenerator](FileMeshGenerator.md)
```

then the `ParsedElementIDMeshGenerator` adds three eeid **(a, b, c)** to the mesh. The values aginst eeid **(a, b, c)** will be **(-1, 44, 20)** through out the whole mesh.

> ```

Reviewer 2: For all documentation, make sure to link against files in the repo itself. This will insulate you from future changes to syntax by automatically rendering up to date documentation. For example, if you want to display a snippet of `my_file.i` to show just the `[Mesh]` block, it would be:

```
!listing test/tests/my_file.i
block=Mesh
```

For example, you can take a look at the documentation for existing objects in Cardinal to see how the input file snippets get displayed. https://cardinal.cels.anl.gov/source/userobjects/HexagonalSubchannelBin.html

[Mesh]
 [file_mesh_generator]
    type=FileMeshGenerator
    input=mesh_in.e
 []
  [add_eeid_block]
    type = ParsedElementIDMeshGenerator
    input = file_mesh_generator
    extra_element_integer_names = 'a b c'
    values = '-1 44 20'
  []
[]

> ```

Reviewer 2: Follow the pattern for the other mesh generator documentation. You'll want to have this at the end of the file:

```
!syntax parameters /Mesh/ParsedElementIDMeshGenerator

!syntax inputs /Mesh/ParsedElementIDMeshGenerator
```

This will render in the documentation an up-to-date list of (i) all of the input files in the Cardinal repository that use this object and (ii) the object's valid parameters and the short string descriptions of what they mean. You can check out, for example, the bottom of this page: https://cardinal.cels.anl.gov/source/userobjects/HexagonalSubchannelBin.html
